processing/src/test/java/org/apache/druid/segment/transform/TransformerTest.java

```diff
@@ -385,7 +389,7 @@ public void testTransformWithArrayLongInputs()
     Assert.assertNotNull(actual);
     Assert.assertEquals(ImmutableList.of("dim"), actual.getDimensions());
     Assert.assertArrayEquals(new Object[]{1L, 2L, null, 3L}, (Object[]) actual.getRaw("dim"));
-    Assert.assertEquals(Arrays.asList("1", "2", "null", "3"), actual.getDimension("dim"));
+    Assert.assertArrayEquals(new String[]{"1", "2", null, "3"}, actual.getDimension("dim").toArray());
```

Comment:
nit: any reason to convert to array? I think `Arrays.asList` works with nulls, it was probably a bug earlier that this was being converting `null` to `"null"`

Reply:
assertEquals found it as other reference of ArrayList and failed with assertion 
```
java.lang.AssertionError: expected: java.util.Arrays$ArrayList<[1.2000000476837158, 2.299999952316284, null, 3.4000000953674316]> but was: java.util.ArrayList<[1.2000000476837158, 2.299999952316284, null, 3.4000000953674316]>
Expected :java.util.Arrays$ArrayList<[1.2000000476837158, 2.299999952316284, null, 3.4000000953674316]>
Actual   :java.util.ArrayList<[1.2000000476837158, 2.299999952316284, null, 3.4000000953674316]>
```